sensor/common/detector/networkbaseline/network_baseline.go

```diff
@@ -70,7 +71,24 @@ func (e *networkBaselineEvaluator) checkPeerInBaselineForEntity(
 		// If no baseline exists then we do not mark it as anomalous
 		return true
 	}
-	peer := networkbaseline.PeerFromNetworkEntityInfo(peerEntity, peerEntityName, dstPort, protocol, isIngressToBaselineEntity)
+
+	var peer networkbaseline.Peer
+	if peerEntity.GetType() == storage.NetworkEntityInfo_EXTERNAL_SOURCE &&
+		peerEntity.GetExternalSource().GetDiscovered() {
+		// If a peer is a discovered external entity, we must anonymize it
+		// to the Internet, as the baselines do not currently store or monitor IP
+		// information.
+		peer = networkbaseline.Peer{
+			IsIngress: isIngressToBaselineEntity,
+			DstPort:   dstPort,
+			Protocol:  protocol,
+			Name:      networkgraph.InternetExternalSourceName,
+			Entity:    networkgraph.InternetEntity(),
+		}
```

Comment:
Shouldn't this go to `networkbaseline.PeerFromNetworkEntityInfo`?

Reply:
yeah, fair point, I'll fix this

Reply:
I've had to add an extra helper to differentiate between `storage.NetworkEntityInfo` and `networkgraph.Entity` but it's a bit clearer now I think